python/cells.cpp

```diff
@@ -0,0 +1,226 @@
+    levels.push_back({0});
+
+    // Standard mersenne_twister_engine seeded with gid.
+    std::mt19937 gen(gid);
+    std::uniform_real_distribution<double> dis(0, 1);
+
+    double dend_radius = 0.5; // Diameter of 1 μm for each cable.
+
+    unsigned nsec = 1;
+    for (unsigned i=0; i<params.max_depth; ++i) {
+        // Branch prob at this level.
+        double bp = interp(params.branch_probs, i, params.max_depth);
+        // Length at this level.
+        double l = interp(params.lengths, i, params.max_depth);
+        // Number of compartments at this level.
+        unsigned nc = std::round(interp(params.compartments, i, params.max_depth));
+
+        std::vector<unsigned> sec_ids;
+        for (unsigned sec: levels[i]) {
+            for (unsigned j=0; j<2; ++j) {
+                if (dis(gen)<bp) {
+                    sec_ids.push_back(nsec++);
+                    auto dend = cell.add_cable(sec, arb::section_kind::dendrite, dend_radius, dend_radius, l);
+                    dend->set_compartments(nc);
+                    dend->add_mechanism("pas");
+                    dend->rL = 100;
+                }
+            }
+        }
+        if (sec_ids.empty()) {
+            break;
+        }
+        levels.push_back(sec_ids);
+    }
+
+    // Add spike threshold detector at the soma.
+    cell.add_detector({0,0}, 10);
+
+    // Add a synapse to the mid point of the first dendrite.
+    cell.add_synapse({1, 0.5}, "expsyn");
+
+    // Add additional synapses that will not be connected to anything.
+    for (unsigned i=1u; i<params.synapses; ++i) {
+        cell.add_synapse({1, 0.5}, "expsyn");
+    }
+
+    return cell;
+}
+
+void register_cells(pybind11::module& m) {
+    using namespace pybind11::literals;
+
+    pybind11::class_<arb::spike_source_cell> spike_source_cell(m, "spike_source_cell",
+        "A spike source cell, that generates a user-defined sequence of spikes that act as inputs for other cells in the network.");
+
+    spike_source_cell
+        .def(pybind11::init<>(
+            [](const regular_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at regular intervals")
+        .def(pybind11::init<>(
+            [](const explicit_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at a sequence of user-defined times")
+        .def(pybind11::init<>(
+            [](const poisson_schedule_shim& sched){
+                return arb::spike_source_cell{sched.schedule()};}),
+            "schedule"_a, "Construct a spike source cell that generates spikes at times defined by a Poisson sequence")
+        .def("__repr__", [](const arb::spike_source_cell&){return "<spike_source_cell>";})
+        .def("__str__",  [](const arb::spike_source_cell&){return "<spike_source_cell>";});
```

Comment:
- Add full stops in constructor descriptions. 
- Add `arbor: ` in `__str__` and `__repr__` (return strings)

Reply:
:+1: